app/jobs/import_twilio_data_job.rb

```diff
@@ -0,0 +1,25 @@
+class ImportTwilioDataJob < ActiveJob::Base
+  class InvalidDateParameter < StandardError
+    def initialize(msg, backtrace = nil)
+      set_backtrace(backtrace) if backtrace
+      super(msg)
+    end
+  end
+
+  queue_as :default
+
+  rescue_from(InvalidDateParameter) do |exception|
+    Bugsnag.notify(exception)
+  end
+
+  def perform(date_string)
+    date = parse_date(date_string)
+    DailyCalls::Twilio::Importer.new.import(start_date: date, end_date: date)
+  end
+
+  def parse_date(date_string)
+    Date.parse(date_string)
```

Comment:
You're right that bubbling `ArgumentError` isn't such a great idea. Perhaps we could reduce this to just rescue `ArgumentError` right here, and calling  `Bugsnag.notify` with the wrapped exception - changing the message to include the `date_string` if necessary? This way we could drop the exception hook that feels a teeny bit overkill since we're controlling where `InvalidDataParameter` is raised (just here basically).